classes/class04/homework/routes/base.js

```diff
@@ -0,0 +1,99 @@
+var schema = require('../models/schema');
+
+exports.home = function (req, res) {
+    res.render('index');
+};
+
+exports.ingredients = function (req, res) {
+    schema.Ingredient.find({}, function (err, data) {
+        res.render('ingredients', {ingredients: data});
+    })
+}
+
+exports.order = function (req, res) {
+    schema.Ingredient.find({}, function (err, data) {
+        if (err) {
+            console.log("Failed to get list of ingredients");
+            res.status(500, {error: "Failed to get list of ingredients " + err});
+            return;
+        }
+        res.render('order', {ingredients: data});
+    });
+}
+
+exports.kitchen = function (req, res) {
+    schema.Order.find({}, function (err, data) {
+        res.render("kitchen", {orders: data});
+    });
+}
+
+exports.addOrder = function (req, res) {
+    var ingredients = [];
+    for (var key in req.body) {
+        if (key && key != "total_cost" && key != "name" && key != "ingredientStr" && key != req.body.hasOwnProperty(key) && req.body[key]) {
```

Comment:
Given the DOM context (of the form) available on the client you are able to do the parsing in a more easily understandable way. Once you pack all the data into your request, the server no longer has the original context and your parsing is much more difficult to understand.